common/flatpak-transaction-log.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright © 2017 Endless Mobile, Inc.
+ *
+ * This file is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *       Sam Spilsbury <sam@endlessm.com>
+ */
+
+#include "config.h"
+
+#include <json-glib/json-glib.h>
+
+#include "flatpak-transaction-log.h"
+
+struct _FlatpakTransactionLog {
+  GObject    parent;
+
+  GFile      *path;
+};
+
+G_DEFINE_TYPE (FlatpakTransactionLog, flatpak_transaction_log, G_TYPE_OBJECT)
+
+typedef enum {
+  PROP_PATH = 1,
+  LAST_PROP
+} FlatpakTransactionLogProperty;
+
+/**
+ * flatpak_transaction_log_new:
+ * @path: a #GFile indicating where the transaction log will be created
+ *
+ * Creates a new #FlatpakTransactionLog object. This class encapsulates
+ * a special append-only JSON-like log file indicating the transactions
+ * that have been performed by flatpak over time. The log file created or
+ * appended to at @path does not conform to RFC7159, instead it is a
+ * newline delimited list of JSON-formatted object definitions which should
+ * be parsed one line at a time. This enables #FlatpakTransactionLog to write
+ * new entries in O(1) time, as opposed to having to read the entire log
+ * into memory so that it can be re-written again.
```

Comment:
The JSON schema isn’t documented anywhere. Even if you don’t write an actual [JSON Schema](http://json-schema.org/) file at the moment, it would be useful to list the keys and types supported in a gtk-doc comment somewhere. (Maybe in the gtk-doc comment for the overall `FlatpakTransactionLog` class?)